datafusion/optimizer/src/simplify_expressions/expr_simplifier.rs

```diff
@@ -365,6 +367,32 @@ impl<'a, S: SimplifyInfo> ExprRewriter for Simplifier<'a, S> {
                     None => lit_bool_null(),
                 }
             }
+            Expr::InList {
+                expr: _expr,
+                list,
+                negated,
+            } if list.is_empty() => lit(negated),
+
+            Expr::InList {
+                expr,
+                list,
+                negated,
+            } if list.len() <= THRESHOLD_INLINE_INLIST && expr.try_into_col().is_ok() => {
```

Comment:
I recommend also adding an explicit case for `list.len() == 1` (without `expr.try_into_col()`)